src/datasets/features/audio.py

```diff
@@ -174,7 +174,7 @@ def _decode_mp3(self, path_or_file):
 
         array, sampling_rate = torchaudio.load(path_or_file, format="mp3")
         if self.sampling_rate and self.sampling_rate != sampling_rate:
-            if not hasattr(self, "_resampler"):
+            if not hasattr(self, "_resampler") or self._resampler.orig_freq != sampling_rate:
```

Comment:
Nice workaround, this should minimize the number of required `_resampler` replacements :+1: